src/test/java/tests/TurboIssueEventTests.java

```diff
@@ -0,0 +1,400 @@
+package tests;
+
+import static org.junit.Assert.assertEquals;
+import github.IssueEventType;
+import github.TurboIssueEvent;
+
+import java.util.ArrayList;
+import java.util.GregorianCalendar;
+import java.util.List;
+
+import org.eclipse.egit.github.core.User;
+import org.junit.Test;
+
+public class TurboIssueEventTests {
+
+    public List<TurboIssueEvent> sampleEvents;
+
+    public static TurboIssueEvent createLabelUpdateEvent(
+            String userName, IssueEventType eventType,
+            GregorianCalendar time, String labelName) {
+
+        return new TurboIssueEvent(
+                   new User().setLogin(userName), eventType, time.getTime())
+               .setLabelName(labelName);
+    }
+
+    public TurboIssueEventTests() {
+        sampleEvents = new ArrayList<>();
+
+        sampleEvents.add(
+                createLabelUpdateEvent("A", IssueEventType.Labeled,
+                        new GregorianCalendar(2015, 1, 1, 1, 1, 0),
+                        "A1"));
+        sampleEvents.add(
+                createLabelUpdateEvent("A", IssueEventType.Unlabeled,
+                        new GregorianCalendar(2015, 1, 1, 1, 2, 0),
+                        "A2"));
+        sampleEvents.add(
+                createLabelUpdateEvent("B", IssueEventType.Labeled,
+                        new GregorianCalendar(2015, 1, 1, 1, 0, 30),
+                        "B1"));
+        sampleEvents.add(
+                createLabelUpdateEvent("B", IssueEventType.Labeled,
+                        new GregorianCalendar(2015, 1, 1, 1, 1, 0),
+                        "B2"));
+        sampleEvents.add(
+                createLabelUpdateEvent("B", IssueEventType.Unlabeled,
+                        new GregorianCalendar(2015, 1, 1, 1, 1, 31),
+                        "B1"));
+        sampleEvents.add(
+                createLabelUpdateEvent("C", IssueEventType.Labeled,
+                        new GregorianCalendar(2015, 1, 1, 2, 30, 15),
+                        "C1"));
+        sampleEvents.add(
+                createLabelUpdateEvent("D", IssueEventType.Unlabeled,
+                        new GregorianCalendar(2015, 1, 1, 3, 20, 59),
+                        "D1"));
+    }
+
+    /**
+     * The method groupLabelUpdateEvents should not modify the input events list
+     */
+    @Test
+    public void testLabelUpdateEventsGroupingInvariant() {
+        List<TurboIssueEvent> events = new ArrayList<>(sampleEvents);
+        TurboIssueEvent.groupLabelUpdateEvents(events);
+        assertEquals(sampleEvents, events);
+    }
+
+    /**
+     * Test the events grouping when list of events is empty
+     */
+    @Test
+    public void testLabelUpdateEventsGroupingForNoEvent() {
+        assertEquals(new ArrayList<TurboIssueEvent>(),
+                     TurboIssueEvent.groupLabelUpdateEvents(new ArrayList<TurboIssueEvent>()));
+    }
+
+    /**
+     * Test the events grouping when there is only 1 event
```

Comment:
Test**s** the events ... (it's a shorter way of saying 'This method tests the events ...).